### PR TITLE
Fix complex plugin dependencies parsing.

### PR DESCRIPTION
**Pull request by manthey**
If plugins A and B are enabled, and plugin A depends on plugin C, and B depends on C and D, and D depends on E, the dependencies of plugin D are never analyzed if plugin A is examined before B (and, hence, plugin E won't be added).

Also, when the plugins are toggled via settings, log why other plugins are enabled.

### CODE REVIEW

**Comment by manthey** [comment]
The important change is just converting a `return` to a `continue` in a routine that was last touched by @zachmullen two years ago.

File: girder/utility/plugin_utilities.py [hunk 1]
```diff
@@ -122,7 +122,7 @@ def loadPlugins(plugins, root, appconf, apiRoot=None, buildDag=True):
     return root, appconf, apiRoot
 
 
-def getToposortedPlugins(plugins=None, ignoreMissing=False, keys=('dependencies',)):
```
Comment (danlamanna) [inline girder/utility/plugin_utilities.py]: Is there a reason `logNew` has to exist at all? Can we just always add the info log message?

File: girder/utility/plugin_utilities.py [hunk 2]
```diff
@@ -122,7 +122,7 @@ def loadPlugins(plugins, root, appconf, apiRoot=None, buildDag=True):
...
+def getToposortedPlugins(plugins=None, ignoreMissing=False, keys=('dependencies',), logNew=False):
```
Comment (danlamanna) [inline girder/utility/plugin_utilities.py]: Is there a reason `logNew` has to exist at all? Can we just always add the info log message?

**Comment by danlamanna** [comment]
Can you configure your editor to [wrap](https://chris.beams.io/posts/git-commit/#wrap-72) your commit messages?

File: girder/utility/plugin_utilities.py (hunk 2 above)
Comment (brianhelba) [inline girder/utility/plugin_utilities.py]: Agreed, I think we should not clutter the business logic API with logging options. I think the best practice is to log everything (sending it to `DEBUG` level if you're concerned about spam under the default configuration) then control the output via filters in handlers.

**Comment by brianhelba** [comment]
@manthey LGTM

**Pull request merged**